pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -38,9 +43,18 @@ import (
 	"istio.io/istio/pkg/workloadapi"
 )
 
-// AmbientIndex maintains an index of ambient WorkloadInfo objects by various keys.
+type AmbientIndex interface {
```

Comment:
Do all interface member methods lock? Might be nice to comment which ones do lock so consumers don't unknowingly add lock pressure in a loop or something

Reply:
Yes, they all lock. I've added docs to the methods.